### PR TITLE
Extending EPB documentation added.

### PR DESCRIPTION
**Pull request by AMZN-scspaldi**
Signed-off-by: scspaldi <scspaldi@amazon.com>

### CODE REVIEW

File: editorpythonbindings.md [hunk 1]
```diff
@@ -0,0 +1,568 @@
+---
+description: ' Follow these steps to use or extend EditorPythonBindings. '
+title: Using & Extending EditorPythonBindings
+---
+
+EditorPythonBindings (EPB) offers access to build scalable multi-point, scriptable, end-to-end, round-trip pipelines, and common framework solutions for game content creators and Open 3D engine (O3DE) contributors via Python. These solutions will measurably drive things like:
```
Comment (evanchia-ly-sdets) [inline editorpythonbindings.md:6]: I think "engine" should be capitalized

File: editorpythonbindings.md [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+
+* Improve extensibility by reducing the barrier of entry for building out new custom workflows or pipelined processes
+* Increase efficiency via automation to reduce the number of clicks to operate
+* Allow automated content to process and run hands-free
+
+## Using EPB with Tests
+While the O3DE public documentation should serve as the primary source for usage, here is a brief example of how to use EPB in a test. 
+
+This code snippet the entry point into the entire object/widget hierarchy of our Qt application.  From here, you can do things like access the menu bars/toolbars, find children widgets to inspect/inject events onto, and more.
```
Comment (evanchia-ly-sdets) [inline editorpythonbindings.md:15]: Missing word at start of sentence

File: editorpythonbindings.md [hunk 3]
```diff
@@ -0,0 +1,568 @@
...
+
+```python
+import Tests.ly_shared.pyside_utils as pyside_utils
```
Comment (evanchia-ly-sdets) [inline editorpythonbindings.md:18]: Have we moved this module to the editor test tools package?

File: editorpythonbindings.md [hunk 4]
```diff
@@ -0,0 +1,568 @@
...
+# Returns a PySide2 wrapped instance of our QMainWindow
+editor_window = pyside_utils.get_editor_main_window()
+```
+
+As you can see, you can treat EPB as you can most other Python modules.
+
+## Extending EPB
+
+### Overview
+The EPB gem is an optional gem. This means that when a project (e.g. Starter Game or Samples Project) wants to expose scriptable functionality to Python, a project team needs to enable the “Editor Python Bindings” gem for the project. The next step is for the feature team to add behavior scoped for automation.
```
Comment (evanchia-ly-sdets) [inline editorpythonbindings.md:29]: Do we still have StarterGame and SamplesProject? I thought those were deprecated.

**Review by evanchia-ly-sdets — CHANGES_REQUESTED** [review]

File: editorpythonbindings.md [hunk 5]
```diff
@@ -0,0 +1,566 @@
...
+    }
+}
+```
+
+##### Scoping For In Game Mode
+There are actually two separate game client modes 1) the game Launcher run time and 2) the game mode inside the Editor. The Python scripts will not work in the first client mode since the Python VM is not loaded in the Launcher.
+
+The second mode can run Python script code but it is not fully supported; so a feature team can use Python code in the Editor's game mode but not all behavior has been set up to work in this game mode.
+
+The default Scope value is AZ::Script::Attributes::ScopeFlags::Launcher  since the scripting contexts for Lua and Scripting Canvas where the first mechanisms for binding behavior.
```
Comment (jackalbe) [inline editorpythonbindings.md:94]: The AZ::Script::Attributes::ScopeFlags::Common and AZ::Script::Attributes::ScopeFlags::Automation flags should also be mentioned

File: editorpythonbindings.md [hunk 6]
```diff
@@ -0,0 +1,566 @@
...
+EditorPythonBindings (EPB) offers access to build scalable multi-point, scriptable, end-to-end, round-trip pipelines, and common framework solutions for game content creators and Open 3D Engine (O3DE) contributors via Python. These solutions will measurably drive things like:
+
+* Improve extensibility by reducing the barrier of entry for building out new custom workflows or pipelined processes
+* Increase efficiency via automation to reduce the number of clicks to operate
+* Allow automated content to process and run hands-free
+
+## Using EPB with Tests
+While the O3DE public documentation should serve as the primary source for usage, here is a brief example of how to use EPB in a test. 
+
+This code snippet is the entry point into the entire object/widget hierarchy of our Qt application.  From here, you can do things like access the menu bars/toolbars, find children widgets to inspect/inject events onto, and more.
```
Comment (jackalbe) [inline editorpythonbindings.md:15]: The "QtForPython" gem is also required for Qt PySide2 stuff.

**Review by evanchia-ly-sdets — CHANGES_REQUESTED** [review]

File: editorpythonbindings.md [hunk 7]
```diff
@@ -0,0 +1,568 @@
...
+            AZStd::string_view filename,
+            const AZStd::vector<AZStd::string_view>& args);
+    };
+    using EditorPythonRunnerRequestBus = AZ::EBus<EditorPythonRunnerRequests>;
+  
+} // namespace AzToolsFramework
+```
+
+## Related  Documentation
+<!-- TODO: Include link to O3DE EPB documentation -->
```
Comment (evanchia-ly-sdets) [inline editorpythonbindings.md:568]: Just a reminder for this TODO

**Review by evanchia-ly-sdets — APPROVED** [review]

**Review by jackalbe — APPROVED** [review]

**Pull request merged**